DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/AdminFieldsController.cs

```diff
@@ -146,6 +147,107 @@ BulkAdminFieldAnswersViewModel model
             );
         }
 
+        [HttpGet]
+        [Route("{customisationId:int}/AdminFields/Add/New")]
+        public IActionResult AddAdminFieldNew(int customisationId)
+        {
+            TempData.Clear();
+
+            var model = new AddAdminFieldViewModel(customisationId);
+            var addAdminFieldData = new AddAdminFieldData(model);
+            var id = addAdminFieldData.Id;
+
+            Response.Cookies.Append(
+                AddPromptCookieName,
+                id.ToString(),
+                new CookieOptions
+                {
+                    Expires = CookieExpiry
+                }
+            );
+            TempData.Set(addAdminFieldData);
+
+            return RedirectToAction("AddAdminField", new { customisationId = model.CustomisationId });
+        }
+
+        [HttpGet]
+        [Route("{customisationId:int}/AdminFields/Add")]
+        public IActionResult AddAdminField(int customisationId)
+        {
+            var addAdminFieldData = TempData.Peek<AddAdminFieldData>()!;
+
+            SetViewBagCoursePromptNameOptions(addAdminFieldData.AddModel.CustomPromptId);
+
+            var model = addAdminFieldData?.AddModel ?? new AddAdminFieldViewModel(customisationId);

```

Comment:
When would we get here without the temp data?

Reply:
I've removed the `?? new AddAdminFieldViewModel(customisationId);` part and added a service filter. Though when I navigate to /Add without clicking through the button, I get redirected to /AdminFields rather than /TrackingSystem/CourseSetup/24286/AdminFields and I'm not sure why?